lib/maths/CDataFrameCategoryEncoder.cc

```diff
@@ -231,8 +231,12 @@ std::size_t CEncodedDataFrameRowRef::numberColumns() const {
     return m_Encoder->numberEncodedColumns();
 }
 
-CDataFrameCategoryEncoder::CDataFrameCategoryEncoder(CMakeDataFrameCategoryEncoder builder) {
-    m_Encodings = builder.makeEncodings();
+CDataFrameCategoryEncoder::CDataFrameCategoryEncoder(CMakeDataFrameCategoryEncoder builder)
+    : CDataFrameCategoryEncoder(&builder) {
+}
+
+CDataFrameCategoryEncoder::CDataFrameCategoryEncoder(CMakeDataFrameCategoryEncoder* builder) {
```

Comment:
This seems odd to me: if this is passed in by pointer then it could be null, but this can't support it being null. What is forcing us to create this new constructor?

Reply:
Hm, I need this to override the logic of `CMakeDataFrameCategoryEncoder` for unit testing. To this end, inherit from `CMakeDataFrameCategoryEncoder`, but if I pass it by reference to the encoder, I loose polymorphic behavior. Do you have any better idea?